.github/workflows/pytest.yml

```diff
@@ -0,0 +1,31 @@
+name: Pytest
+
+on:
+  push:
+    branches: gawsalya-pytest
+  pull_request:
+    branches: gawsalya-pytest
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.11
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r ./live_pipeline/requirements.txt
+        pip install -r ./dashboard/requirements.txt
+        pip install -r ./archive_pipeline/requirements.txt
+    - name: Test with pytest
+      run: |
+        pytest
```

Comment:
Really nicely done, clear and concise - great work!

Reply:
Impressive